container/bitvector/bitvector.go

```diff
@@ -31,6 +31,50 @@ func (vector *BitVector) Length() int {
 	return vector.length
 }
 
+// And will AND two BitVector's and returns the resulting one
+func (a *BitVector) And(b *BitVector) *BitVector {
+	vector := NewBitVector(make([]byte, 0), 0)
+	if a.length != b.length {
+		return vector
```

Comment:
Returning an empty bit vector when the lengths are different seems unhelpful.  When the lengths are different, maybe use to longest vector's length instead (and 0 pad the shorter vector).

Same for Or